graph2/graph.proto

```diff
@@ -0,0 +1,307 @@
+	// Sourcegraph server.
+	string html_url = 7 [(gogoproto.customname) = "HTMLURL"];
+
+	// DefaultBranch is the default VCS branch used (typically "master" for git
+	// repositories and "default" for hg repositories).
+	string default_branch = 8;
+
+	// Language is the primary programming language used in this source tree.
+	string language = 9;
+
+	// Blocked is whether this repo has been blocked by an admin (and
+	// will not be returned via the external API).
+	bool blocked = 10;
+
+	// Deprecated source trees are labeled as such and hidden from global search
+	// results.
+	bool deprecated = 11;
+
+	// Fork is whether this source tree is a VCS repository fork.
+	bool fork = 12;
+
+	// Mirror indicates whether this source tree's canonical location is on
+	// another server. Mirror source trees track their upstream.
+	bool mirror = 13;
+
+	// Private is whether this source tree is private.
+	bool private = 14;
+
+	// CreatedAt is when this source tree was created. If it represents an
+	// externally hosted (e.g., GitHub) source tree, the creation date is when it
+	// was created at that origin.
+	pbtypes.Timestamp created_at = 15;
+
+	// UpdatedAt is when this source tree's metadata was last updated (on its origin if
+	// it's an externally hosted source tree).
+	pbtypes.Timestamp updated_at = 16;
+
+	// PushedAt is when this source tree's source code was last updated (e.g., via
+	// VCS-push).
+	pbtypes.Timestamp pushed_at = 17;
+
+	// Permissions describes the permissions that the current user (or anonymous users,
+	// if there is no current user) is granted to this repository.
+	RepoPermissions permissions = 18;
+
+	GitHubRepo github = 19 [(gogoproto.customname) = "GitHub"];
+}
+
+message TreeRev {
+	Tree tree = 1;
+
+	// Rev is the version of the source tree
+	string rev = 2;
+
+	// Type of revision (e.g., commit ID, branch name, or tag for VCS
+	// repositories; version string for build system packages)
+	string rev_type = 3;
+}
+
+// RawDep represents a raw dependency as it is defined in the build
+// system. This should include all the information that the build
+// system needs to fetch the dependency but does not necessarily
+// universally uniquely identify the dependency. For instance, in
+// Python, the raw dependency might be urllib3=0.1.0, but depending on
+// what PyPI index is used, the dependency could resolve to the
+// version of urllib3 hosted on http://pypi.python.org or somewhere
+// else (private codebases often point pip at a private Python package
+// index).
+message RawDep {
+	string name = 1;
+	string version = 2;
+}
+
+// UnitKey universally uniquely identifies a build unit.
+message UnitKey {
+	TreeKey key = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true, (gogoproto.jsontag) = ""];
+
+	// Version is the build-system-level version string. It can also be a commit
+	// ID, but no client code should depend on implied VCS semantics (e.g., never
+	// run `git clone buildUnit.Version`).
```

Comment:
The `git clone` command example doesn't make sense to me - why would someone try to run `git clone` on something that isn't a clone URL anyway?